### DISCUSSION TITLE
Error aunching sl-migration on debian10

### DISCUSSION THREAD

**Discussion by MrWu-at-msv** [simple-login/app#738] [Self-hosting question] [answered]
I am testing the install procedure for self hosting simplelogin and I get this error when I try to run the sl-migration container on Debian 10
the database container is running and working
following files are configured and double checked
> /etc/postfix/main.cf
> /etc/postfix/pgsql-relay-domains.cf
> /etc/postfix/pgsql-transport-maps.cf
> $(pwd)/simplelogin.env

here is the full error, any tip is highly appreciated.

> root@simplelogin:~# lsb_release -a
> No LSB modules are available.
> Distributor ID:	Debian
> Description:	Debian GNU/Linux 10 (buster)
> Release:	10
> Codename:	buster
> 
> >root@simplelogin:~# docker run --rm \
> >     --name sl-migration \
> >     -v $(pwd)/sl:/sl \
> >     -v $(pwd)/sl/upload:/code/static/upload \
> >     -v $(pwd)/dkim.key:/dkim.key \
> >     -v $(pwd)/dkim.pub.key:/dkim.pub.key \
> >     -v $(pwd)/simplelogin.env:/code/.env \
> >     --network="sl-network" \
> >     simplelogin/app:3.4.0 flask db upgrade
> 
> >>> URL: http://app.simplelogin.obfuscateurl.com
> MAX_NB_EMAIL_FREE_PLAN is not set, use 5 as default value
> Paddle param not set
> Upload files to local dir
> >>> init logging <<<
> 2022-01-04 07:16:12,175 - SL - DEBUG - "/code/app/utils.py:11" - <module>() -  - load words file: /code/local_data/words_alpha.txt
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2338, in _wrap_pool_connect
>     return fn()
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 304, in unique_connection
>     return _ConnectionFairy._checkout(self)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 778, in _checkout
>     fairy = _ConnectionRecord.checkout(pool)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 495, in checkout
>     rec = pool._do_get()
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/impl.py", line 239, in _do_get
>     return self._create_connection()
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 309, in _create_connection
>     return _ConnectionRecord(self)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 440, in __init__
>     self.__connect(first_connect_check=True)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 661, in __connect
>     pool.logger.debug("Error on connect(): %s", e)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 69, in __exit__
>     exc_value, with_traceback=exc_tb,
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
>     raise exception
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 656, in __connect
>     connection = pool._invoke_creator(self)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/strategies.py", line 114, in connect
>     return dialect.connect(*cargs, **cparams)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 493, in connect
>     return self.dbapi.connect(*cargs, **cparams)
>   File "/usr/local/lib/python3.7/site-packages/psycopg2/__init__.py", line 127, in connect
>     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
> psycopg2.OperationalError: FATAL:  password authentication failed for user "a5wie356tpur2kbf8djbgg"
> 
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/usr/local/bin/flask", line 8, in <module>
>     sys.exit(main())
>   File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 967, in main
>     cli.main(args=sys.argv[1:], prog_name="python -m flask" if as_module else None)
>   File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 586, in main
>     return super(FlaskGroup, self).main(*args, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 782, in main
>     rv = self.invoke(ctx)
>   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 610, in invoke
>     return callback(*args, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
>     return f(get_current_context(), *args, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/flask/cli.py", line 426, in decorator
>     return __ctx.invoke(f, *args, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/click/core.py", line 610, in invoke
>     return callback(*args, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/flask_migrate/cli.py", line 134, in upgrade
>     _upgrade(directory, revision, sql, tag, x_arg)
>   File "/usr/local/lib/python3.7/site-packages/flask_migrate/__init__.py", line 96, in wrapped
>     f(*args, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/flask_migrate/__init__.py", line 271, in upgrade
>     command.upgrade(config, revision, sql=sql, tag=tag)
>   File "/usr/local/lib/python3.7/site-packages/alembic/command.py", line 298, in upgrade
>     script.run_env()
>   File "/usr/local/lib/python3.7/site-packages/alembic/script/base.py", line 489, in run_env
>     util.load_python_file(self.dir, "env.py")
>   File "/usr/local/lib/python3.7/site-packages/alembic/util/pyfiles.py", line 98, in load_python_file
>     module = load_module_py(module_id, path)
>   File "/usr/local/lib/python3.7/site-packages/alembic/util/compat.py", line 184, in load_module_py
>     spec.loader.exec_module(module)
>   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
>   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
>   File "migrations/env.py", line 96, in <module>
>     run_migrations_online()
>   File "migrations/env.py", line 81, in run_migrations_online
>     with connectable.connect() as connection:
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2265, in connect
>     return self._connection_cls(self, **kwargs)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 104, in __init__
>     else engine.raw_connection()
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2372, in raw_connection
>     self.pool.unique_connection, _connection
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2342, in _wrap_pool_connect
>     e, dialect, self
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1584, in _handle_dbapi_exception_noconnection
>     sqlalchemy_exception, with_traceback=exc_info[2], from_=e
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
>     raise exception
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 2338, in _wrap_pool_connect
>     return fn()
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 304, in unique_connection
>     return _ConnectionFairy._checkout(self)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 778, in _checkout
>     fairy = _ConnectionRecord.checkout(pool)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 495, in checkout
>     rec = pool._do_get()
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/impl.py", line 239, in _do_get
>     return self._create_connection()
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 309, in _create_connection
>     return _ConnectionRecord(self)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 440, in __init__
>     self.__connect(first_connect_check=True)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 661, in __connect
>     pool.logger.debug("Error on connect(): %s", e)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 69, in __exit__
>     exc_value, with_traceback=exc_tb,
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
>     raise exception
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/pool/base.py", line 656, in __connect
>     connection = pool._invoke_creator(self)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/strategies.py", line 114, in connect
>     return dialect.connect(*cargs, **cparams)
>   File "/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 493, in connect
>     return self.dbapi.connect(*cargs, **cparams)
>   File "/usr/local/lib/python3.7/site-packages/psycopg2/__init__.py", line 127, in connect
>     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
> sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) FATAL:  password authentication failed for user "a5wie356tpur2kbf8djbgg"
> 
> (Background on this error at: http://sqlalche.me/e/13/e3q8)
> 
>

**Comment by nguyenkims** [comment]
Hi the error is `psycopg2.OperationalError: FATAL: password authentication failed for user "a5wie356tpur2kbf8djbgg"` which means the container can't connect to the database. I wonder if the database is run in the `sl-network` docker network?

**Comment by MrWu-at-msv** [comment, marked as answer]
TYVM for your help
i found the problem for the auth on the db.
the first time I create the DB with myuser as  POSTGRES_USER
then I removed the sl-db docker and I create it again with the new username
but the postgreeSQL database doesn't have the appropriate Privileges to the newly created users and I couldn't give the privilege to the new user manually

> simplelogin=> \du
>                                           List of roles
>        Role name        |                         Attributes                         | Member of 
> ------------------------+------------------------------------------------------------+-----------
>  a5wie356tpur2kbf8djbgg |                                                            | {}
>  myuser                 | Superuser, Create role, Create DB, Replication, Bypass RLS | {}

GRANT ALL PRIVILEGES ON simplelogin TO a5wie356tpur2kbf8djbgg;
didn't work
I did not try to remove and recreate the a5wie356tpur2kbf8djbgg USER manually.

instead, I wipe the sl directory, recreated the needed subfolders, and recreate the sl-db docker from scratch.

PS: maybe you can track down the issue if someone needs to change the POSTGRES_USER on a running install
